plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -582,25 +583,41 @@ func BuildDimensions(mTags map[string]string) []*cloudwatch.Dimension {
 		})
 	}
 
-	var keys []string
-	for k := range mTags {
-		if k != "host" {
-			keys = append(keys, k)
+	globalDimensionKeys := make([]string, 0, len(globalDimensions))
+
+	for key, value := range globalDimensions {
+		if key != "" && value != "" {
+			globalDimensionKeys = append(globalDimensionKeys, key)
 		}
 	}
-	sort.Strings(keys)
 
-	for _, k := range keys {
-		if len(dimensions) >= MaxDimensions {
-			log.Printf("D! max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number", MaxDimensions, len(dimensions))
-			break
-		}
+	if len(globalDimensionKeys)+len(dimensions) > MaxDimensions {
+		log.Printf("D! max MaxDimensions %v is less than than number of global dimensions plus host dimension (if present) %v thus only taking the max number", MaxDimensions, len(globalDimensionKeys)+len(dimensions))
+		sort.Strings(globalDimensionKeys)
+		globalDimensionKeys = globalDimensionKeys[:MaxDimensions-len(dimensions)]
+	}
 
-		value := mTags[k]
-		if value == "" {
-			continue
+	for _, key := range globalDimensionKeys {
+		dimensions = append(dimensions, &cloudwatch.Dimension{
+			Name:  aws.String(key),
+			Value: aws.String(globalDimensions[key]),
+		})
+	}
+
+	var keys []string
+	for key, value := range mTags {
+		if key != "" && value != "" && globalDimensions[key] == "" {
+			keys = append(keys, key)
 		}
+	}
+
+	if len(dimensions)+len(keys) > MaxDimensions {
+		log.Printf("D! max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number", MaxDimensions, len(dimensions)+len(keys))
+		sort.Strings(keys)
+		keys = keys[:MaxDimensions-len(dimensions)]
+	}
```

Comment:
This change is so that we only bother to sort the keys if we actually need to (i.e. there are too many dimensions)